Content.Server/Ame/AmeNodeGroup.cs

```diff
@@ -177,6 +177,18 @@ public float InjectFuel(int fuel, out bool overloading)
         return powerOutput;
     }
 
+    /// <summary>
+    /// Calculates the amount of power the AME can produce with the given settings
+    /// </summary>
+    /// <remarks>
+    /// We use float because the math will fail if we use integers
+    /// </remarks>
+
+    public float CalculatePower(float fuel, float cores)
```

Comment:
Why does the math "fail if we use integers?" The formal arguments should be integers, because both fuel and number of cores are integers. If there is a cast necessary, cast it here. But because you have the literal float in front, then this calculation should already all be floating.

Reply:
The remark was (and still is) in the code in InjectFuel where the calculation originally took place. (AMENodeGroup Line 129).

We need floats, since the return value is a float . Dividing an integer by an integer would return a rounded integer. (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators#division-operator-)

You are right, the settings values are integers. I'll change the function to take integers and only cast right before calculating. I would prefer having all AME fuel related numbers in floats, but thats a discussion for another time.